### PR TITLE
Removes the docs site and adds signposts for the new docs repo

### PR DESCRIPTION
**Pull request by ctauchen**
With migration complete, we should remove the docs site from this repo. This PR may do the trick. Or it'll help start the discussion about what else needs to be done to remove docs entirely.

What do you think, @projectcalico/core-maintainers?

Needs CP to `release-v3.25`, `release-v3.24`

### CODE REVIEW

**Comment by nelljerram** [comment]
@ctauchen First observation: the 5 docs-related checks that are shown (in "Some checks were not successful") as failed.  I presume we should remove those, for master and for the branches that are now in the new docs home.

File: CONTRIBUTING_DOCS.md [hunk 1]
```diff
@@ -1,291 +1,5 @@
 # Contributing to Calico documentation
 
-## Overview
+We moved the Calico docs site to [a new docs repository](https://github.com/tigera/docs).
```
Comment (caseydavenport) [inline CONTRIBUTING_DOCS.md:3]: Do we have an equivalent source of information in the new repo? Do we want to re-use any of this content?

**Review by caseydavenport — APPROVED** [review]
LGTM - one comment / think to think about

**Comment by caseydavenport** [comment]
@ctauchen you'll actually need to remove the CI checks for the docs as well.

Could you remove this block of code from the Semaphore config: https://github.com/projectcalico/calico/blob/f2b5b7b54cc92eeb03638d0532c26f959542af0f/.semaphore/semaphore.yml.tpl#L664-L676

And then run `make gen-semaphore-yaml` in the repository root?

Looks like there are also some conflicts to be resolved, which suggest a couple of changes that might need to be ported to the new docs site :disappointed:

File: CONTRIBUTING_DOCS.md (hunk 1 above)
Comment (ctauchen) [reply CONTRIBUTING_DOCS.md:3]: We still need to build that out, and I will be reusing some of this content. For now, I just want to direct people to the new repo. In time (not too long!) I'll have the equivalent files built.

**Comment by caseydavenport** [comment]
@ctauchen unfortunately I think you need to rebase on master in order to pick up a CI fix

**Comment by ctauchen** [comment]
> @ctauchen unfortunately I think you need to rebase on master in order to pick up a CI fix, but wait until we fix the node dir dependency on the calico/ dir

Sure thing -- let me know when you're ready and I'll make the updates.

**Comment by caseydavenport** [comment]
@ctauchen should be good for a rebase now 🤞

**Comment by ctauchen** [comment]
@caseydavenport You can merge this now. I'm serving Netlify redirect rules from another source.

File: calico/LICENSE [hunk 2]
```diff
@@ -1,201 +0,0 @@
-Apache License
```
[the remark this answers is not in the document]
Comment (ctauchen) [reply calico/LICENSE:1]: Thanks for spotting this, @remram44. Fix is here: https://github.com/projectcalico/calico/pull/7506.

**Pull request merged**